enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,147 @@
+# Work Executor Group
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposal adds a new field `executorGroupName` to the spec of `ManifestWork` 
+api, to separate the execution of a set of `ManifestWork` resources to the 
+dedicated work agents. The default/legacy work agent will be the default
+executor of the `ManifestWork` resource if its `executorGroupName` is not
+specified.
+
+## Motivation
+
+### Controlling availability (blast-radius) of the work agents
+
+Currently, for each of the managed clusters we will deploy the work agents
+in a single deployment resource which is sharing the availability via "leader-
+election", so there's merely one instance of work agent taking effect at a time.
+Under this pattern, we can't break down the availability of the work agent into
+finer units so that:
+
+- During the upgrade, we can progressively reduce the impact from the agent's 
+  down-time.
+- By grouping the agents according to their use-cases, we can control the 
+  blast-radius so that the unavailability will not propogate between the groups.
+
+### Separate the identity and its held permission for the work agents
+
+There can be different actors using the work api to distribute resource from 
+the hub cluster. Currently the work agent has the cluster-admin access to the
+managed cluster which means that every hub users with write access to the
+work resources will be able to dispatch any resources to the managed cluster.
+By making different groups of work agents mount different service accounts and
```

Comment:
This is a cool use-case.  The goal being to control permissions of the hub users on the spokes?

Reply:
the permission control can be done by introducing a validating admission webhook that authorizes whether a hub user can use a executor with the SubjectAccessReview api. i clarified it in the latest revision, PTAL.